AutomatedLabDefinition/AutomatedLabDefinition.psm1

```diff
@@ -2498,7 +2498,11 @@ process
     {
         $machine.OperatingSystem = $OperatingSystem
     }
-    
+    elseif ($machine.HostType -eq 'VMWare')
+    {
+        $machine.OperatingSystem = $OperatingSystem
```

Comment:
VMWare machines should also use the OS returned by Get-LabAvailableOperatingSystem. I would suggest removing this else branch and adding HostType -eq 'VMWare' to the first if condition: if ($machine.HostType -eq 'HyperV' -or $machine.HostType -eq 'VMWare')